language/move-lang/src/expansion/translate.rs

```diff
@@ -203,15 +222,20 @@ fn module(
     context.address = None
 }
 
-fn set_sender_address(context: &mut Context, loc: Loc, sender: Option<Address>) {
+fn set_sender_address(
+    context: &mut Context,
+    loc: Loc,
+    module_name: &ModuleName,
+    sender: Option<Spanned<Address>>,
+) {
     context.address = Some(match sender {
-        Some(addr) => addr,
+        Some(sp!(_, addr)) => addr,
         None => {
             let msg = format!(
-                "Invalid module declaration. No sender address was given as a command line \
-                 argument. Add one using --{}. Or set the address at the top of the file using \
-                 'address _:'",
-                crate::command_line::SENDER
+                "Invalid module declaration. The module does not have a specified address. \
+                 Either declare it inside of an 'address _ {{' block or declare it with an \
```

Comment:
I think something like `<address>` or a concrete example address might be slightly easier to understand than `_`, but no objections if the underscore convention is already used in other error messages.